### PR TITLE
Consolidate all the PYTORCH_NVFUSER_DISABLE_ options like PYTORCH_NVFUSER_DUMP

### LINKED ISSUE #1584: Unify PYTORCH_NVFUSER_DISABLE_ options
There are `PYTORCH_NVFUSER_DISABLE_INDEX_HOIST` and `PYTORCH_NVFUSER_DISABLE_PREDICATE_ELIMINATION`. Unify them as `PYTORCH_NVFUSER_DISABLE` and allow to choose what to disable like `PYTOCH_NVFUSER_DUMP`.

### PR DESCRIPTION
**Pull request by naoyam**
Added `PYTORCH_NVFUSER_DISABLE` environment variable.

```
Available options:
        fallback, fma, index_hoist, nvtx, predicate_elimination
```

I kept `PYTORCH_NVFUSER_DISABLE_FALLBACK` for compatibility. The other `PYTORCH_NVFUSER_DISABLE_` environment variables are no longer used.

Closes #1584

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/utils.cpp [hunk 1]
```diff
@@ -185,27 +228,18 @@ bool isDebugDumpEnabled(DebugDumpOption option) {
...
 }
 
-bool disableIndexHoisting() {
-  const static char* disable_index_hoist =
-      getenv("PYTORCH_NVFUSER_DISABLE_INDEX_HOIST");
-  return disable_index_hoist ? atoi(disable_index_hoist) : false;
-}
+bool useFallback() {
+  // Keep this env var for compatibility
+  const char* disable_fb_env = getenv("PYTORCH_NVFUSER_DISABLE_FALLBACK");
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/utils.cpp:239]: Nitpick, we can throw a warning here for deprecation, if user are still using `disable_fb_env`. Something like
```TORCH_WARN_ONCE("env variable PYTORCH_NVFUSER_DISABLE_FALLBACK is deprecated, use PYTORCH_NVFUSER_DISABLE=\"fallback\" instead")```;

File: torch/csrc/jit/codegen/cuda/README.md [hunk 2]
```diff
@@ -224,7 +224,7 @@ There're a few debug dump that could be turned on via environment variables. Loo
 
 1. There's regression after turning on nvfuser.
 
-First thing is to check that you have fusion kernel running properly. Try to run your model with fallback disabled to see if you hit any errors that caused fallback via `export PYTORCH_NVFUSER_DISABLE_FALLBACK=1`.
+First thing is to check that you have fusion kernel running properly. Try to run your model with fallback disabled to see if you hit any errors that caused fallback via `export PYTORCH_NVFUSER_DISABLE=fallback`.
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/README.md:227]: nitpick: Can we have a quick note on debugging/tuning knobs? Maybe only listing the important ones (e.g. fma, fallback) and point people to the code. Either have that in FAQs, or add a section following `Query nvfuser codegen kernels`

**Review by jjsjann123 — APPROVED** [review]
LGTM, much cleaner than keep adding more env variables~~

File: torch/csrc/jit/codegen/cuda/README.md (hunk 2 above)
Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/README.md:227]: I'll add some short note to the README file.

Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/README.md:227]: Thanks looks good enough~

**Pull request merged**